src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSource.cs

```diff
@@ -20,6 +20,7 @@ public partial class DiagnosticListener : System.Diagnostics.DiagnosticSource, S
         public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string, object?>> observer, System.Predicate<string>? isEnabled) { throw null; }
         public override string ToString() { throw null; }
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("The type of object being written to DiagnosticSource cannot be discovered statically.")]
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode("The object parameter might be of a type that might not be available during runtime.")]
```

Comment:
I am sorry, this still does not make sense. If you have an object, its type has to be available at runtme - otherwise you cannot have the object.

Out of the different variants, I think the best one so far has been the message you had here originally: "The type of object being written to DiagnosticSource cannot be discovered statically."

A different variant would be: "DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling.". It follows the pattern that we have used for System.Linq.Queryable.

Reply:
Sorry for not getting the msg correct and thanks for the suggestion